python/ray/data/_internal/logical/operators/all_to_all_operator.py

```diff
@@ -58,6 +58,7 @@ class RandomShuffle(AbstractAllToAll):
     def __init__(
         self,
         input_op: LogicalOperator,
+        num_outputs: int,
```

Comment:
for a shuffle operation, we expect that `num_outputs` is the same as the input dependency's `num_outputs` right? if that value is available, should we try to use that over requiring a specific value from caller?

Reply:
Doesn't `random_shuffle` have a `num_blocks` arg? If the user doesn't specify `num_blocks` in `random_shuffle` it uses the input_dependency's `num_outputs`.

Reply:
starting in ray 2.9, `num_blocks` is deprecated for `random_shuffle` method, so we can omit this i think: 
https://github.com/ray-project/ray/blob/00d605f6e9783215a79435f88131b5752e516764/python/ray/data/dataset.py#L1152-L1154

Reply:
Does the num_block still work though? If it does, shouldn't we remove this when we actually remove the parameter?

Reply:
if i am understanding correctly, since the new code path does not use `num_blocks` (i.e. when we initialize the `RandomShuffle` operator, we do not pass `num_blocks` parameter), it seems that it would have no effect on the execution: https://github.com/ray-project/ray/blob/master/python/ray/data/dataset.py#L1164